### PR TITLE
Posts form and function.

### PR DESCRIPTION
**Pull request by AsmahanM**
Added a form for users to input: title, alt text and a file (along with a submit button).

Started on a post function for posts to be added. Currently redirects back to home in case of an error but may need to be changed.

### CODE REVIEW

File: routes/posts.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const db = require("/database/model.js");
+
+function get(request, response) {
+try {
+    const postsForm = /*html*/
+`
+    <form enctype="multipart/form-data class="Upload File" id="upload-file" action="/post" method="POST">
```
Comment (lascellesabercrombie) [inline routes/posts.js:7]: I think a quote needs to be closed after form-data

File: routes/posts.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <div>
+      <label for="Title">Title:</label>
+      <input type="text" id="title" placeholder="Title" name="title" >
```
Comment (lascellesabercrombie) [inline routes/posts.js:10]: id case should probably match label case (maybe the lowercase version preferable?)

File: routes/posts.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      </div>
+      <br>
+      <label for="alt-text">Alt Text:</label>
+      <input type="text" id="alt_text" placeholder="Alt Text" name="alt_text" >
```
Comment (lascellesabercrombie) [inline routes/posts.js:14]: label-for/input-id mismatch

File: routes/posts.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    response.send(postsForm);
+} catch (error) {
+    console.error(error);
+    response.status(500).send(`<h1>Error</h1>`)
+}
+}
+
+    function post(request, response) {
+        const {title, alt_text, image} = sanitize(request.body);
+        model
```
Comment (lascellesabercrombie) [inline routes/posts.js:34]: I think around here we need to get the users.id to insert into posts as user_id. I think we need the cookies to do this, which aren't set up yet, but maybe worth flagging with a commented line

**Review by AsmahanM — COMMENTED** [review]
Updated and made changes to posts file

File: layout.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+function layout(title, content) {
+    return /*html*/ `<!DOCTYPE html>
+  <html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="description" content="an app for sharing quotations">
+    <title>${"Upload File"}</title>
```
Comment (lascellesabercrombie) [inline layout.js:8]: This is a placeholder, so should just be "title", which can then be an argument used for generating new pages (e.g. layout('signup', '<p>sign up</p>'))

File: layout.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    <link rel="stylesheet" href="styles.css">
+  </head>
+  <body>
+    ${postsForm}
```
Comment (lascellesabercrombie) [inline layout.js:12]: See comment to line 8

**Pull request merged**